api/videostream_test.go

```diff
@@ -0,0 +1,262 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/ausocean/openfish/api/ds_client"
+	"github.com/gofiber/fiber/v2"
+)
+
+func setup() (*fiber.App, int64) {
+	// Initialize app.
+	ds_client.Init(false)
+	app := fiber.New()
+	RegisterAPIRoutes(app)
+
+	// Populate datastore with capturesource for testing.
+	// Send test request.
+	body := `{
+		"name": "Ochre Point",
+		"location": "-35.221389,138.46333333",
+		"camera_hardware": "rpi cam v2 - wide angle lens",
+		"site_id": 10192840284
+	}`
+	req := httptest.NewRequest("POST", "/api/v1/capturesources", strings.NewReader(body))
+	req.Header.Set("Content-Type", "application/json")
+	res, err := app.Test(req, -1)
+	if err != nil || res.StatusCode != 200 {
+		log.Fatal("Test setup failed.")
+	}
+
+	resBody, err := io.ReadAll(res.Body)
+	if err != nil {
+		log.Fatal("Test setup failed.")
```

Comment:
It might be better if this function takes a t *testing.T instead and then use t.Fatalf. Ditto for other places in this setup function where you're calling log.Fatal

Reply:
Agreed